lib/lint.ml

```diff
@@ -277,21 +465,42 @@ let msg_of_error (package, (err : Checks.error)) =
          a bug in dune (https://github.com/ocaml/dune/issues/2147) this should \
          never be the case. Please remove the {build} tag from its filter."
         pkg
+  | RestrictedPrefix prefix ->
+      Printf.sprintf "Warning in %s: package name has restricted prefix '%s'"
+        pkg prefix
+  | PrefixConflictClassMismatch mismatch ->
+      Checks.Prefix.msg_of_prefix_conflict_class_mismatch ~pkg mismatch
+  | NameCollision other_pkg ->
+      Printf.sprintf "Warning in %s: Possible name collision with package '%s'"
+        pkg other_pkg
+  | UnexpectedFile file ->
+      Printf.sprintf "Error in %s: Unexpected file in %s/%s" pkg
+        (path_from_pkg ~repo_dir:"" package)
+        file
```

Comment:
Perhaps for a followup: do you think we should exit with different codes depending on a warning vs. error? Or perhaps accept a flag that lets us exit with 1 on a warning, but exit with a 0 by default?

Reply:
Yes, that might be useful, especially if we are going the "use the CLI in CI" route.

Reply:
https://github.com/tarides/infrastructure/issues/342